activerecord/lib/active_record/connection_adapters/abstract/transaction.rb

```diff
@@ -533,7 +553,7 @@ def within_new_transaction(isolation: nil, joinable: true)
         @connection.lock.synchronize do
           transaction = begin_transaction(isolation: isolation, joinable: joinable)
           begin
-            yield
+            yield transaction
```

Comment:
Should we check arity here? Otherwise `Article.transaction` blocks actually must accept an arg, which is an API break, no. Is that intended?